scripts/exglobal_atmos_products.sh

```diff
@@ -0,0 +1,269 @@
+    if (( last > ncount )); then (( last = ncount )); fi
+
+    # if final record of is u-component, add next record v-component
+    # if final record is land, add next record icec
+    # grep returns 1 if no match is found, so temporarily turn off exit on non-zero rc
+    set +e
+    # shellcheck disable=SC2312
+    ${WGRIB2} -d "${last}" "${tmpfile}" | grep -E -i "ugrd|ustm|uflx|u-gwd|land"
+    rc=$?
+    set_strict
+    if (( rc == 0 )); then  # Matched the grep
+      last=$(( last + 1 ))
+    fi
+    if (( iproc == nproc )); then
+      last=${ncount}
+    fi
+
+    # Break tmpfile into processor specific chunks in preparation for MPMD
+    ${WGRIB2} "${tmpfile}" -for "${first}":"${last}" -grib "${tmpfile}_${iproc}"
+    export err=$?; err_chk
+    input_file="${tmpfile}_${iproc}"
+    output_file_prefix="pgb2${grp}file_${fhr3}_${iproc}"
+    echo "${INTERP_ATMOS_MASTERSH} ${input_file} ${output_file_prefix} ${grid_string}" >> "${DATA}/poescript"
+
+    # if at final record and have not reached the final processor then write echo's to
+    # poescript for remaining processors
+    if (( last == ncount )); then
+      for (( pproc = iproc+1 ; pproc < nproc ; pproc++ )); do
+        echo "/bin/echo ${pproc}" >> "${DATA}/poescript"
+      done
+      break
+    fi
+  done  # for (( iproc = 1 ; iproc <= nproc ; iproc++ )); do
+
+  # Run with MPMD or serial
+  if [[ "${USE_CFP:-}" = "YES" ]]; then
+    "${HOMEgfs}/ush/run_mpmd.sh" "${DATA}/poescript"
+    export err=$?
+  else
+    chmod 755 "${DATA}/poescript"
+    bash +x "${DATA}/poescript" 2>&1 mpmd.out
+    export err=$?
+  fi
+  err_chk
+
+  # We are in a loop over downset, save output from mpmd into nset specific output
+  cat mpmd.out  # so we capture output into the main logfile
+  mv mpmd.out "mpmd_${nset}.out"
+
+  # Concatenate grib files from each processor into a single one
+  # and clean-up as you go
+  echo "Concatenating processor-specific grib2 files into a single product file"
+  for (( iproc = 1 ; iproc <= nproc ; iproc++ )); do
+    for grid in "${grids[@]}"; do
+      cat "pgb2${grp}file_${fhr3}_${iproc}_${grid}" >> "pgb2${grp}file_${fhr3}_${grid}"
+      rm  "pgb2${grp}file_${fhr3}_${iproc}_${grid}"
+    done
+    # There is no further use of the processor specific tmpfile; delete it
+    rm "${tmpfile}_${iproc}"
+  done
+
+  # Move to COM and index the product grib files
+  for grid in "${grids[@]}"; do
+    prod_dir="COM_ATMOS_GRIB_${grid}"
+    ${NCP} "pgb2${grp}file_${fhr3}_${grid}" "${!prod_dir}/${PREFIX}pgrb2${grp}.${grid}.${fhr3}"
+    ${WGRIB2} -s "pgb2${grp}file_${fhr3}_${grid}" > "${!prod_dir}/${PREFIX}pgrb2${grp}.${grid}.${fhr3}.idx"
+  done
+
+  # Create supplemental 1-degree grib1 output TODO: who needs 1-degree grib1 product?
+  # move to COM and index it
+  if (( nset == 1 )); then
+    if [[ "${PGBS:-}" == "YES" ]]; then
+      if [[ "${PGB1F:-}" == "YES" ]]; then
+        ${CNVGRIB} -g21 "pgb2${grp}file_${fhr3}_1p00" "pgb${grp}file_${fhr3}_1p00"
```

Comment:
On a quick test, this line is returning 0 statuses, but with several `putgbexn error =         8` messages.  Tracked this down to the w3emc module w3fi73.f, which fails when searching through bitmap fields.

Reply:
The log file can be found here: /work/noaa/global/dhuber/para/com/2110/logs/2021110818/gdasatmprod_f000-f003.log.

Reply:
Is this the case in `develop` as well unless the `pgb2file_fHHH_1p00` from this PR is not identical to `develop`.

Reply:
Not sure if it is in `develop`, but it was not in my `spack-stack` tests.  I can run a `develop` test if you like.

Reply:
I can run the reference with `develop` and cross-check.

Reply:
@DavidHuber-NOAA Following the comment from @WenMeng-NOAA I have removed `pgrb.1p00` generation.  There is no file, ergo, the error is no longer relevant.  We should still spot-check and compare the `pgrb2` files between develop and this branch.